extensions/GitExtension/FileExplorerGitIntegration/Models/GitCommandRunner.cs

```diff
@@ -0,0 +1,229 @@
+        log.Information("Git Exe Install Path stored successfully");
+        return true;
+    }
+
+    public bool DetectGit()
+    {
+        var gitExeFound = false;
+
+        if (!gitExeFound)
+        {
+            // Check if git.exe is present in PATH environment variable
+            try
+            {
+                System.Diagnostics.Process.Start("git.exe");
+                gitExeFound = true;
+                StoreGitExeInstallPath("git.exe");
+            }
+            catch (Exception ex)
+            {
+                log.Debug(ex, "Failed to start Git.exe");
+            }
+        }
+
+        if (!gitExeFound)
+        {
+            // Check execution of git.exe by finding install location in registry keys
+            string[] registryPaths = { "HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Git_is1", "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Git_is1" };
+
+            foreach (var registryPath in registryPaths)
+            {
+                var gitPath = Registry.GetValue(registryPath, "InstallLocation", defaultValue: string.Empty) as string;
+                if (!string.IsNullOrEmpty(gitPath))
+                {
+                    var paths = FindSubdirectories(gitPath);
+                    CheckForExeInPaths(paths, ref gitExeFound);
+                    if (gitExeFound)
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (!gitExeFound)
+        {
+            // Search for git.exe in common file paths
+            var programFiles = Environment.GetEnvironmentVariable("ProgramFiles");
+            var programFilesX86 = Environment.GetEnvironmentVariable("ProgramFiles(x86)");
+            string[] possiblePaths = { $"{programFiles}\\Git\\bin", $"{programFilesX86}\\Git\\bin", $"{programFiles}\\Git\\cmd", $"{programFilesX86}\\Git\\cmd" };
+            CheckForExeInPaths(possiblePaths, ref gitExeFound);
+        }
+
+        return gitExeFound;
+    }
+
+    private string[] FindSubdirectories(string installLocation)
+    {
+        try
+        {
+            if (Directory.Exists(installLocation))
+            {
+                return Directory.GetDirectories(installLocation);
+            }
+            else
+            {
+                log.Warning("Install location does not exist: {InstallLocation}", installLocation);
+                return Array.Empty<string>();
+            }
+        }
+        catch (Exception ex)
+        {
+            log.Warning(ex, "Failed to find subdirectories in install location: {InstallLocation}", installLocation);
+            return Array.Empty<string>();
+        }
+    }
+
+    private void CheckForExeInPaths(string[] possiblePaths, ref bool gitExeFound)
+    {
+        foreach (var path in possiblePaths)
```

Comment:
NIT: possiblePaths.Where(x => !string.IsNullOrEmpty(x))

Reply:
Resolved in latest. Thanks!